### ISSUE TITLE
The used google contacts API is no longer supported

### ISSUE THREAD

**Issue by cubei** [jitsi/jitsi#129]
OAuth 1.0 was [depricated for 3 years and is now no longer supported](https://developers.google.com/google-apps/contacts/v3/). With the current implementation access to google contacts is no longer possible.

[See mail](http://lists.jitsi.org/pipermail/users/2015-May/009470.html)

Windows
- Jitsi 2.8.5426
- Guava: Google Core Libraries for Java 15.0.0
- User search contact source 0.0.1

**Comment by geotree** [comment]
Confirmed : google contacts feature in Jitsi stopped working a few days ago :-(

**Comment by cubei** [comment]
It's working again for some reason

**Comment by cubei** [comment]
I don't know why google reactivated the API for a moment, but now it's broken again.

**Comment by cobratbq** [comment]
We are looking into this. For completeness, we were not using OAuth 1.0 for accessing the Google Contacts API, but Client Login - which we did use - is also deprecated and removed/disabled. So, yes we need an alternative.

**Comment by cubei** [comment]
Any news on this defect? Jitsi is unusable like this, because it doesn't support contact import from csv or something like that (please correct me if I'm wrong). So there is no way to get my contacts in there => unusable.

**Comment by cubei** [comment]
Has Jitsi no users or how is it working for them?
If Jitsi would support CSV import: no problem. But this is just ridicules.
2 Month and no way to get my contacts into Jitsi.

**Comment by damencho** [comment]
We are working on the google contacts api problem. You can try using the addressbook contact sources like Thindebird, Outlook or Mac OSX adressbook, I think you can import the contacts in most of them.

**Comment by cobratbq** [comment]
The implementation is mostly complete. The issue at this point is to provide a decent user interface so you can input the approval code you have received from Google. This is a new way of authentication not previously used in the Google Contacts plugin, so we need to create some of the pieces.

**Comment by cubei** [comment]
Thanks for the replies, gives me hope. Good to hear you're working on it, I appreciate it. Sorry for beeing a little harsh ;)
Google contacts -> Thunderbird -> Jitsi Import works, so take your time =)

**Comment by cobratbq** [comment]
Initial implementation is commited, should appear in next nightly. It may have some rough edges but should work.

**Comment by cubei** [comment]
I installed the nightly from 2015-07-22 jitsi-2.9.5460-x64.
When I want to search for a contact, I have to input my password
![jitsi](https://cloud.githubusercontent.com/assets/2197875/8834060/5a2260e0-30b2-11e5-8cf6-f2cf7affc862.png)
But this window pops up for every search. So if I want to search for "test". I type 't' and type in my password- > type 'e' and type in my password -> 's' ... -> 't'
That's a "little" annoying, because the password is a app specific password for 2 step authentication, so I usually should generate it once and forget it.

But the most important part is: It's not working, non of the gmail contacts are in the search results. And there is no error message (I guess that is the rough edge you mentioned), I can leave the password field empty or type what ever I want, nothing happens.

**Comment by cobratbq** [comment]
Hi,
      
      Did you enter your user account's password, or did you enter the
      code you get after going to the approval URL?
      To be clear, you cannot use your normal password for this. You
      need to enter an approval URL received from Google when you visit
      the URL by clicking on the "Approve Jitsi's access to
      <account>" link.
      
      This may be a bit unclear. I have received some feedback with
      suggestions on how to improve. For now, please see my email in
      dev-list or I'll post the relevant bit below.
      
      Danny
      
      
      However, OAuth 2 works slightly different and Jitsi depends on 1
additional step from the user. The first time Google Contacts is used
for a Google account in Jitsi, the user will have to approve Jitsi for
access to the contacts of your account. (Once for every configured account.)

So how does that work?
1. Configure your Google account as you would normally. (If you did this
already, then you need not repeat this step.)
2. On first usage, Jitsi will show a popup asking you to visit a URL by
clicking the provided link. This link takes you to a Google URL at which
Google will ask you to approve a Jitsi for access to your contacts. This
will allow Jitsi to download profile data for your contacts.
PLEASE NOTE that you need to approve the account that is named in the
popup window. You may need to log in with the correct Google account for
this.
3. After approving, you will go to a new website on which Google
provides you with an access code. This code needs to be copied to the
Jitsi popup dialog. Once confirmed, Jitsi will get access to the
account. A token is stored (safely, like passwords) for future use.

Every next time you will use Google Contacts, Jitsi will automatically
request access using the token it received earlier. Unless you lose your
Jitsi configuration, you should not need to approve access anymore. It
is also possible to revoke Jitsi's access. You can find this option in
the Security options of your Google account. (Log in to Google.)
      
      
      
      
      On 22-07-15 20:50, Lars Harmsen wrote:
    
    
      
      I installed the nightly from 2015-07-22 jitsi-2.9.5460-x64.
        When I want to search for a contact, I have to input my password
        
        But this window pops up for every search. So if I want to search
        for "test". I type 't' and type in my password- > type 'e'
        and type in my password -> 's' ... -> 't'
        That's a "little" annoying, because the password is a app
        specific password for 2 step authentication, so I usually should
        generate it once and forget it.
      But the most important part is: It's not working, non of the
        gmail contacts are in the search results. And there is no error
        message (I guess that is the rough edge you mentioned), I can
        leave the password field empty or type what ever I want, nothing
        happens.
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by cubei** [comment]
Awesome. Thanks, perfect instructions.

**Comment by cubei** [comment]
Awesome. Thanks, perfect instructions.

So, it's working. Thanks for the support and work.

**Comment by cendp1am** [comment]
I'm on Ubuntu Linux, Jitsi 2.8.5426, I added my google account, created the 'app specific' password for Jitsi access to my google account, entered in the Jitsi configuration. I didn't get any pop message asking for 'code'.  any suggestions?
warm thanks!
peter

**Comment by cubei** [comment]
@cendp1am Yes, the version you are using is the stable build, that is over a year old, the fix is not part of it. You should use a nightly.

**Comment by cendp1am** [comment]
Thanks @cubei. I installed, but have problem with my Java and Jitsi exits while saying 'Exception in thread "main" java.lang.UnsupportedClassVersionError: net/java/sip/communicator/launcher/SIPCommunicator : Unsupported major.minor version 51.0'. I tried installing also jdk version 7 simultaneously to jdk 6, but it didn't help. I'll not fiddle around my Ubuntu 12.04 LTS, cause It'd make me more troubles probably. I'm keeping such old Ubuntu version for stability and will only update next year maybe.
peter
